compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -1176,6 +1182,107 @@ class ResolverImpl(
         }
         return modifiers
     }
+
+    private enum class RefPosition {
+        PARAMETER_TYPE,
+        RETURN_TYPE,
+        SUPER_TYPE
+    }
+
+    // TODO: Strict mode for catching unhandled cases.
+    private fun findRefPosition(ref: KSTypeReference): RefPosition = when (val parent = ref.parent) {
+        is KSCallableReference -> when (ref) {
+            parent.returnType -> RefPosition.RETURN_TYPE
+            else -> RefPosition.PARAMETER_TYPE
+        }
+        is KSFunctionDeclaration -> when (ref) {
+            parent.returnType -> RefPosition.RETURN_TYPE
+            else -> RefPosition.PARAMETER_TYPE
+        }
+        is KSPropertyGetter -> RefPosition.RETURN_TYPE
+        is KSPropertyDeclaration -> when (ref) {
+            parent.type -> RefPosition.RETURN_TYPE
+            else -> RefPosition.PARAMETER_TYPE
+        }
+        is KSClassDeclaration -> RefPosition.SUPER_TYPE
+        // is KSTypeArgument -> RefPosition.PARAMETER_TYPE
+        // is KSAnnotation -> RefPosition.PARAMETER_TYPE
+        // is KSTypeAlias -> RefPosition.PARAMETER_TYPE
+        // is KSValueParameter -> RefPosition.PARAMETER_TYPE
+        // is KSTypeParameter -> RefPosition.PARAMETER_TYPE
+        else -> RefPosition.PARAMETER_TYPE
+    }
+
+    private fun KSTypeReference.isReturnTypeOfAnnotationMethod(): Boolean {
+        var candidate = this.parent
+        while (this !is KSClassDeclaration && candidate != null)
+            candidate = candidate.parent
+        return candidate.safeAs<KSClassDeclaration>()?.classKind == ClassKind.ANNOTATION_CLASS
```

Comment:
it looks like this is about a type reference inside an annotation class declaration, should it be named annotation method in that sense? I thought annotation method should means a use-site annotation rather than declaration site.

Reply:
Yes, this is checking whether it is a return type / property type in a annotation class declaration. Do you mean change the name or it should check use-site instead?